servers/visual/rasterizer.cpp

```diff
@@ -561,3 +561,98 @@ int RasterizerStorage::multimesh_get_visible_instances(RID p_multimesh) const {
 AABB RasterizerStorage::multimesh_get_aabb(RID p_multimesh) const {
 	return _multimesh_get_aabb(p_multimesh);
 }
+
+Rect2 RasterizerCanvas::Item::calculate_polygon_bounds(const Item::CommandPolygon &p_polygon) const {
+	Rect2 r;
+	int l = p_polygon.points.size();
+	const Point2 *pp = &p_polygon.points[0];
+
+	// First add all untransformed points to the bound.
+	// This will be final version if no skeleton.
+	r.position = pp[0];
+	for (int j = 1; j < l; j++) {
+		r.expand_to(pp[j]);
+	}
+
+	// If there is a skeleton, some or all of the points may be transformed
+	// by bones (depending on the weights).
+	// If all points are fully weighted by bones, this means the original point is not needed for the bound,
+	// so this routine will OVER-ESTIMATE the bound with skeletons. Perhaps this can be improved.
+	if (skeleton != RID()) {
+		// calculate bone AABBs
+		int bone_count = RasterizerStorage::base_singleton->skeleton_get_bone_count(skeleton);
+		ERR_FAIL_COND_V(bone_count < 0, Rect2());
+
+		Rect2 *bptr = nullptr;
+		LocalVector<Rect2> bone_aabbs;
+		if (bone_count <= 1024) {
+			bptr = (Rect2 *)alloca(sizeof(Rect2) * bone_count);
+		} else {
+			bone_aabbs.resize(bone_count);
+			bptr = bone_aabbs.ptr();
+		}
+
+		for (int j = 0; j < bone_count; j++) {
+			bptr[j].size = Vector2(-1, -1); //negative means unused
+		}
+		if (l && p_polygon.bones.size() == l * 4 && p_polygon.weights.size() == p_polygon.bones.size()) {
+			// We need to take into account the skeleton base transform
+			// in case it has been offset by the user.
+			Transform2D skel_trans_inv = RasterizerStorage::base_singleton->skeleton_get_base_transform_2d(skeleton).affine_inverse();
+			Transform2D item_transform = skel_trans_inv * xform; // Transform into bone space.
+			Transform2D item_transform_inv = item_transform.affine_inverse(); // Transform back to local space.
```

Comment:
Note this is equivalent (no need to `affine_inverse()` twice):
```suggestion
			Transform2D skel_trans = RasterizerStorage::base_singleton->skeleton_get_base_transform_2d(skeleton);
			Transform2D item_transform_inv = xform.affine_inverse() * skel_trans;
```